gather.go

```diff
@@ -386,18 +386,23 @@ func (a *Agent) gatherCandidatesRelay(urls []*URL, wg *sync.WaitGroup) error {
 					return locConn.Close()
 				},
 			}
-			candidate, err := NewCandidateRelay(&relayConfig)
-			if err != nil {
+			relayConnClose := func() {
 				if relayConErr := relayConn.Close(); relayConErr != nil {
 					a.log.Warnf("Failed to close relay %v", relayConErr)
 				}
+			}
+			candidate, err := NewCandidateRelay(&relayConfig)
+			if err != nil {
+				relayConnClose()
 
 				client.Close()
 				closeConnAndLog(locConn, a.log, fmt.Sprintf("Failed to create relay candidate: %s %s: %v\n", network, raddr.String(), err))
 				return
 			}
 
 			if err := a.addCandidate(candidate, relayConn); err != nil {
```

Comment:
`Agent.run()` fails after closing the agent. In this case, relay conn is not closed.